File: web/src/services/report.ts
import { http } from './http'

export type ReportItem = {
  id: number
  tradeDate: string
  stockCode: string
  stockName: string
  summary: string
  riskLevel: string
  recommendation: string
}

export async function fetchReports(): Promise<ReportItem[]> {
  const response = await http.get('/reports')
  return response.data.data.items
}

File: backend/internal/service/report.go
package service

import "context"

type IReport interface {
	List(ctx context.Context) ([]ReportItem, error)
}

type ReportItem struct {
	ID             uint   `json:"id"`
	TradeDate      string `json:"tradeDate"`
	StockCode      string `json:"stockCode"`
	StockName      string `json:"stockName"`
	Summary        string `json:"summary"`
	RiskLevel      string `json:"riskLevel"`
	Recommendation string `json:"recommendation"`
}

var localReport IReport

func Report() IReport {
	return localReport
}

func RegisterReport(s IReport) {
	localReport = s
}
